### ISSUE TITLE
Improve S4144: ignore numerical and string literals 

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarTS#599]
[RSPEC-4144](https://jira.sonarsource.com/browse/RSPEC-4144) has been updated to ignore numerical and string literals. Thus such case should raise an issue:
```ts
class A {
    fnOne() {
        this.foo = this.given + 1;
        this.fnFoo("xXx", this.foo);
        this.fooArray.push(1);
        this.isFoo = true;
        this.fnFoo("yYy", this.foo);
    }

    fnTwo() {
        this.foo = this.given + 2;
        this.fnFoo("xXx", this.foo);
        this.fooArray.push(2);
        this.isFoo = true;
        this.fnFoo("yYy", this.foo);
    }
}
```

Fixes #596
